templates/_helpers.tpl

```diff
@@ -623,3 +623,20 @@ Inject extra environment populated by secrets, if populated
 {{ "https" }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+imagePullSecrets generates pull secrets from either string or map values.
+A map value must be indexable by the key 'name'.
+*/}}
+{{- define "imagePullSecrets" -}}
+{{- with .Values.global.imagePullSecrets -}}
+imagePullSecrets:
+{{- range . -}}
+{{- if typeIs "string" . }}
+  - name: {{ . }}
```

Comment:
@benashz shouldn't be just `- {{ . }}` ?

cc @tvoran

Reply:
@chukka for `imagePullSecrets` we need a mapping of `name: <value>`, so in the case of string values we need to create the mapping  dynamically from the string value.